### ISSUE TITLE
Discuss compilers

### ISSUE THREAD

**Issue by ghost** [LoungeCPP/loungesome-cpp#3]
Does MSVC belong in the list? Does it?

To be fair, mingw is complete garbage if you try to build a windows program.

**Comment by Morwenn** [comment]
I have been running MinGW-w64 for years now almost without a problem (I'm looking at you `std::random_device`). The biggest problem I faced was the lack of precompiled libraries.

**Comment by ghost** [comment]
But you're mostly writing console apps, right? When I said "windows program" I really meant linking directx, wic, mfc, etc.

**Comment by Morwenn** [comment]
Oh, like actually _windowed_ programs using Windows-specific libraries. Then I don't know, the only windowed programs that I happened to compile with MinGW-w64 used either SDL or Qt.

**Comment by nabijaczleweli** [comment]
I had no problems linking to native Windows windowed libraries using MinGW

**Comment by kbok** [comment]
Nab, which libs did you use?

-----Original Message-----

**Comment by nabijaczleweli** [comment]
Fuck if I know, 'twas a long time ago I last made a GUI program

**Comment by ghost** [comment]
So we put mingw in the list? It's worthwhile?

IMO the list should contain: Visual Community, clang, g++.

**Comment by nabijaczleweli** [comment]
What's a "Visual Community"?

**Comment by ghost** [comment]
It's a community of people that you can see (i.e. not ghosts, ninjas, or harry potter)

**Comment by nabijaczleweli** [comment]
@kbok what's the _VIsual Community_ you were referring to your previous comment?

**Comment by Morwenn** [comment]
IMO, MinGW-w64 is worth adding to the compiler list because it still works pretty well. MSVC is also of interest because it still manages to implement some features (modules, coroutines...) before other compilers and plays a somewhat important part in the C++ ecosystem.

**Comment by ghost** [comment]
I agree. MSVC has its unique features that makes it interesting, and the debugger is very nice. Worthwhile.

**Comment by nabijaczleweli** [comment]
lol MSVC, the only role it plays in the ecosystem is it puts back global warming ever so slightly (oh, and additional person-hours for making workarounds for it).

**Comment by nabijaczleweli** [comment]
I also propose `rustc`.

**Comment by ghost** [comment]
What's rustc?

**Comment by nabijaczleweli** [comment]
@kbok [this](https://www.google.pl/search?q=rustc&ie=utf-8&oe=utf-8&gws_rd=cr&ei=HFmiVqfXFaaeywOXp6fAAg)

**Comment by milleniumbug** [comment]
-1 because it's not a C++ compiler

(inb4 neither is MSVC)

**Comment by nabijaczleweli** [comment]
Your inb4 sums it up really well, @milleniumbug

**Comment by TheCatPlusPlus** [comment]
> Does MSVC belong in the list? Does it?

lol no (alt: I can't hear you over the sound of my two-phase lookup)

> To be fair, mingw is complete garbage if you try to build a windows program.

Then don't include it. I thought this was supposed to be a repository of quality tools, not just any tools from given category that happen to work (sometimes).

Plus everyone knows all of those goddamn compilers, what's the point of including literally all of them. I wouldn't bother with any of them if there was a choice.

**Comment by Morwenn** [comment]
inb4 catsome-cpp.

**Comment by nabijaczleweli** [comment]
plz no cimploler warz

**Comment by ghost** [comment]
I disagree

There's quite a lot of noobs out there that don't know about clang or msvc, or don't know that it's free. I think it's helpful.

**Comment by Morwenn** [comment]
It's more about the links to useful pages. For example, downloading an actually usable version of MinGW-w64 is a bit tricky when you want to use the compiler for the first time.

**Comment by milleniumbug** [comment]
@kbok I think we should finally ask ourselves a very important question: "what is the scope of this project?". Since this is a lounge project I thought this is a list of C++ tools/libraries with a Lounge Seal of Quality. In this case, listing MSVC would be [very ironic](http://chat.stackoverflow.com/search?q=MSVC&Room=10&page=1&pagesize=50&sort=stars). Furthermore it doesn't fit our acceptance criteria we agreed on. 

List of all the things C++ related already exists, remember why we started this.

**Comment by TheCatPlusPlus** [comment]
Is this a tutorial for people who have no idea about the basic C++ ecosystem or is this a collection of "actually decent, actually used AND appreciated in the industry C++ libraries" which is your project description btw.

**Comment by kbok** [comment]
I don't see how those should be mutually exclusive.

-----Original Message-----

**Comment by Morwenn** [comment]
That said we could also change the description by « we didn't like the awesome-cpp list so we made our own ».

**Comment by nabijaczleweli** [comment]
@kbok Thing is, "tutorial for people who have no idea about the basic C++ ecosystem" is not included in the first place, so exclusivity of those two doesn't come into play anyway.

**Comment by milleniumbug** [comment]
I feel another lounge project incoming...

**Comment by nabijaczleweli** [comment]
And make Katt write it
